src/game/server/gamemodes/base_pvp/base_pvp.cpp

```diff
@@ -780,8 +782,8 @@ int CGameControllerPvp::OnCharacterDeath(class CCharacter *pVictim, class CPlaye
 	if(GameServer()->GetDDRaceTeam(pVictim->GetPlayer()->GetCid()))
 		return 0;
 
-	if(pKiller == pVictim->GetPlayer())
-		pVictim->GetPlayer()->DecrementScore(); // suicide or world
+	if(SuicideOrWorld)
+		pVictim->GetPlayer()->DecrementScore();
```

Comment:
This is not about sprees is it? What is your motivation here?

Reply:
Code deduplication and less function calls :P